docs/en/getting-started/get-started-stack.asciidoc

```diff
@@ -0,0 +1,871 @@
+sudo service logstash start
+----------------------------------------------------------------------
+
+*mac:*
+
+["source","sh",subs="attributes,callouts"]
+----------------------------------------------------------------------
+./bin/logstash -f logstash.conf
+----------------------------------------------------------------------
+
+*win:*
+
+["source","sh",subs="attributes,callouts"]
+----------------------------------------------------------------------
+bin\logstash.bat -f logstash.conf
+----------------------------------------------------------------------
+
+NOTE: The default configuration for {beats} and {ls} uses plain TCP. For
+encryption you must explicitly enable SSL when you configure {beats} and {ls}.
+
+{ls} reads from the {beats} input and indexes events into {es}. You haven't
+defined a filter section yet, so {ls} simply forwards events to {es} without
+additional processing. Next, you learn how to define the filter stage.
+
+[float]
+[[logstash-filter]]
+==== Define a filter to extract data from a field
+
+The system metrics collected by {metricbeat} include a field called `cmdline`
+that contains the full command-line argument used to start the process. For
+example:
+
+[source,json]
+----
+"cmdline": "/Applications/Firefox.app/Contents/MacOS/plugin-container.app/Contents/MacOS/plugin-container -childID 3
+-isForBrowser -boolPrefs 36:1|299:0| -stringPrefs 285:38;{b77ae304-9f53-a248-8bd4-a243dbf2cab1}| -schedulerPrefs
+0001,2 -greomni /Applications/Firefox.app/Contents/Resources/omni.ja -appomni
+/Applications/Firefox.app/Contents/Resources/browser/omni.ja -appdir
+/Applications/Firefox.app/Contents/Resources/browser -profile
+/Users/dedemorton/Library/Application Support/Firefox/Profiles/mftvzeod.default-1468353066634
+99468 gecko-crash-server-pipe.99468 org.mozilla.machname.1911848630 tab"
+----
+
+
+You might decide that you don't want to send the whole command-line argument to
+{es}. You can write a grok filter that matches the first part of the command
+and saves it to a new field. You can also remove the original field, `cmdline`,
+from the event. For example:
+
+// REVIEWERS: I took a stab at this, but I'm not very good at grok or
+// regex. In fact, I sort of hate it...a lot. Can someone review and suggest
+// something better based on the format you would expect in this field?
+
+[source,ruby]
+----
+filter {
+  if [system][process] {
+    if [system][process][cmdline] {
+      grok {
+        match => { "[system][process][cmdline]" => ["%{CMD:[system][process][short_cmdline]}"] }
+        pattern_definitions => {
+          "CMD" => "(?:/[A-Za-z0-9$.+!*'(){},~:;=@#%&_\-]*)+\s*-*[A-Za-z0-9$.+!*'(){},~:;=@#%&_\-]*"
+        }
+        remove_field => "[system][process][cmdline]"
+      }
+    }
+  }
+}
+----
+
+
+Learning grok is beyond the scope of this tutorial, but if you want to learn
+more, see the {logstash-ref}/plugins-filters-grok.html[Grok filter plugin]
+documentation.
```

Comment:
Maybe this should be moved up sooner?  I hit the first occurrence of "grok filter" in the paragraph above the example and went down a rabbit hole trying to figure out whether I was supposed to know what a grok filter was.

Reply:
Grok can be exceptionally powerful, but it's a super deep rabbit hole to go down before developing solid footing on what Filters bring to the table. Maybe we should start with a different filter?

In the Logstash community forums, we tend to steer people _away_ from Grok until they've exhausted other avenues for extracting data (e.g., often-overlooked tools like the Dissect filter can be significantly simpler to implement and maintain). And, because Grok is a pattern language, it is _necessary_ to start with a corpus of messages; starting with a single message is generally a recipe for [disfluency](https://en.wikipedia.org/wiki/Garden_path_sentence).

Reply:
For now, I've updated the tutorial to reflect your suggestions for the current example. We can always merge and improve later, if necessary.